### PR TITLE
chore(vue): migrate to standardized tsconfig and align exports test

### PR DESCRIPTION
**Pull request by calebpollman**
#### Description of changes
Update `@aws-amplify/ui-vue` TypeScript configuration and exports test to align with other packages

#### Issue #, if available
NA

#### Description of how you validated changes
- smoke tested local builds
- ran linting
- ran unit tests
- manually ran example apps

#### Checklist

- [x] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [x] PR description included
- [x] `yarn test` passes and tests are updated/added

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/configs/typescript/tsconfig.vue.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "extends": "./tsconfig.base.json",
+  "compilerOptions": {
+    // required in `vue` projects
+    "jsx": "preserve",
+    // see: https://github.com/vuejs/test-utils/issues/1538
+    "skipLibCheck": true,
+    "types": ["vite/client"]
```
Comment (calebpollman) [inline packages/configs/typescript/tsconfig.vue.json:8]: `jsx`, `skipLibCheck` and `types` are Vue specific. The rest of the properties in the current `ui-vue` [_tsconfig.json_](https://github.com/aws-amplify/amplify-ui/blob/main/packages/vue/tsconfig.json) can be extended from the shared base [_tsconfig.base.json_](https://github.com/aws-amplify/amplify-ui/blob/main/packages/configs/typescript/tsconfig.base.json) or were extraneous

File: packages/vue/__tests__/exports.spec.ts [hunk 2]
```diff
@@ -1,9 +1,10 @@
-import * as exported from '../dist';
+import * as exported from '../src';
```
Comment (calebpollman) [inline packages/vue/__tests__/exports.spec.ts:1]: Import from `../src` to align with other Amplify UI packages

File: packages/vue/jest.config.ts [hunk 3]
```diff
@@ -23,8 +23,11 @@ const config: Config = {
   },
   testEnvironment: 'jsdom',
   verbose: true,
-  moduleFileExtensions: ['js', 'ts', 'json', 'vue', 'tsx'],
-  moduleNameMapper: { '^nanoid$': '<rootDir>/../../node_modules/nanoid' },
+  moduleFileExtensions: ['js', 'ts', 'vue'],
+  moduleNameMapper: {
+    '\\.css$': '<rootDir>/__mocks__/styleMock.js',
```
Comment (calebpollman) [inline packages/vue/jest.config.ts:28]: Added to appease `jest` when parsing CSS imports

File: packages/vue/package.json [hunk 4]
```diff
@@ -39,7 +39,6 @@
   },
   "dependencies": {
     "@aws-amplify/ui": "6.0.8",
-    "@vue/tsconfig": "^0.1.3",
```
Comment (calebpollman) [inline packages/vue/package.json:42]: Unused dependency

**Review by dbanksdesign — APPROVED** [review]

**Review by esauerbo — APPROVED** [review]

**Pull request merged**